### PR TITLE
DAOS-2676 test: skip rebuild test if pool create fails

### PR DESCRIPTION
**Pull request by wangdi1**
Skip rebuild test if pool create fails due to
rank failure.

Signed-off-by: Wang Di <di.wang@intel.com>

### CODE REVIEW

File: src/pool/srv_util.c [hunk 1]
```diff
@@ -451,3 +451,29 @@ ds_pool_check_failed_replicas(struct pool_map *map, d_rank_list_t *replicas,
 	}
 	return 0;
 }
+
+
+int ds_pool_get_ranks_by_uuid(const uuid_t pool_uuid, int status,
+			      d_rank_list_t *ranks)
```
Comment (liuxuezhao) [inline src/pool/srv_util.c]: seems better to comment that user need to do map_ranks_init() for succeed call.

Or change the API to be 
int ds_pool_get_ranks_by_uuid(const uuid_t pool_uuid, int status,
			      d_rank_list_t **ranks);
that do map_ranks_init/map_ranks_fini(), + d_rank_list_dup() internally, and caller need to do d_rank_list_free().

File: src/mgmt/srv_pool.c [hunk 2]
```diff
@@ -41,21 +40,9 @@ ds_mgmt_tgt_pool_destroy(uuid_t pool_uuid, crt_group_t *grp)
...
-		ABT_rwlock_unlock(pool->sp_lock);
-		if (rc != 0) {
-			D_ERROR(DF_UUID": failed to create rank list: %d\n",
-				DP_UUID(pool->sp_uuid), rc);
-			return rc;
-		}
-	}
+	rc = ds_pool_get_ranks_by_uuid(pool_uuid, MAP_RANKS_DOWN, &excluded);
+	if (rc)
+		return rc;
```
Comment (liw) [inline src/mgmt/srv_pool.c]: This is more of temporary hack, I guess? This and the pool create path suffer from the misunderstanding that the MS is part of every pool. That is not the case.

File: src/pool/srv_util.c (hunk 1 above)
Comment (wangdi1) [reply src/pool/srv_util.c]: ah, ok

File: src/mgmt/srv_pool.c (hunk 2 above)
Comment (wangdi1) [reply src/mgmt/srv_pool.c]: Sorry, what do you mean MS here?  Hmm, this is just for function generation.  Though I am not sure exclude the ranks should be done by caller or callee during pool destroy.

File: src/pool/srv_util.c [hunk 3]
```diff
@@ -451,3 +451,29 @@ ds_pool_check_failed_replicas(struct pool_map *map, d_rank_list_t *replicas,
 	}
 	return 0;
 }
+
+/** The caller are responsible for freeing the ranks */
```
Comment (liuxuezhao) [inline src/pool/srv_util.c]: better to be "freeing the ranks by map_ranks_fini()", as the d_rank_list_free() cannot use

File: src/mgmt/srv_pool.c (hunk 2 above)
Comment (gnailzenh) [reply src/mgmt/srv_pool.c]: MS == Management service

**Pull request merged**